src/Commands/viewProperties.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { IotDpsClient, IotDpsModels } from "azure-arm-deviceprovisioningservices";
+import { createAzureClient, IActionContext, openReadOnlyJson } from "vscode-azureextensionui";
+import { Constants } from "../constants";
+import { DpsTreeItem } from "../Treeview/dpsTreeItem";
+import { ExtensionVariables } from "../Utility/extensionVariables";
+
+export async function viewProperties(context: IActionContext, node?: DpsTreeItem): Promise<void> {
+    if (!node) {
+        node = await ExtensionVariables.dpsExtTreeDataProvider.showTreeItemPicker<DpsTreeItem>("IotDps", context);
+    }
+
+    const client: IotDpsClient = createAzureClient(node.root, IotDpsClient);
+    let matchResult = Constants.DpsResourceGroupNameRegex.exec(node.fullId);

```

Comment:
when is fullId set?

Reply:
The azureextensionui package handles the initialization of treeview nodes. FullId is initialized by the package.